structopt-derive/src/lib.rs

```diff
@@ -129,31 +129,62 @@ fn sub_type(t: &syn::Ty) -> Option<&syn::Ty> {
     }
 }
 
-fn extract_attrs<'a>(attrs: &'a [Attribute]) -> Box<Iterator<Item = (&'a Ident, &'a Lit)> + 'a> {
-    let iter = attrs.iter()
+#[derive(Debug, Clone, Copy)]
+enum AttrSource { Struct, Field, }
+
+fn extract_attrs<'a>(attrs: &'a [Attribute], attr_source: AttrSource) -> Box<Iterator<Item = (Ident, Lit)> + 'a> {
```

Comment:
I was pretty close to making this a `Box<Iterator<Item = (Cow<'a, Ident>, Cow<'a, Lit>)>` but luckily I noticed early enough that this was derive macro whose performance will probably never matter 😅